src/Entities/ExerciseResult.cc

```diff
@@ -0,0 +1,46 @@
+#include "ExerciseResult.hh"
+
+#include <optional>
+
+namespace Entities
+{
+    Test::Test(const std::string &name, bool success)
+        : name_(name)
+    {
+        if (success)
+        {
+            errorMessage_ = std::nullopt;
+        }
+        else
+        {
+            errorMessage_ = "";
+        }
+    }
+
+    Test::Test(const std::string &name, const std::string &errorMessage)
+        : name_(name)
+    {
+        this->errorMessage_ = errorMessage;
+    }
```

Comment:
```suggestion
    Test::Test(const std::string &name, const std::string &errorMessage)
        : name_(name),
        , errorMessage_(errorMessage)
    {}
```

Does this work? If not it would still be preferable to call the optional constructor in the initializer list.
Additional note: `this` is probably not needed here.